server/src/utils/database.ts

```diff
@@ -68,7 +68,22 @@ export function searchAssetBuilder(
   builder.andWhere(_.omitBy(id, _.isUndefined));
 
   if (options.userIds) {
-    builder.andWhere(`${builder.alias}.ownerId IN (:...userIds)`, { userIds: options.userIds });
+    if (options.withSharedAlbums && !options.isNotInAlbum) {
+      builder
+        .leftJoin(`${builder.alias}.albums`, 'albums')
+        .leftJoin(`albums.albumUsers`, 'albumUsers')
```

Comment:
These are one-to-many relationships, so you can't do these joins without duplicating rows and messing with pagination. There are other ways to write it, like using `where exists()` or adding a CTE that you do an inner join on.

Reply:
thanks